src/features/apiDelegation/offered/ChooseApiPage/ChooseApiPage.tsx

```diff
@@ -63,6 +61,39 @@ export const ChooseApiPage = () => {
     dispatch(search(''));
   }, []);
 
+  useEffect(() => {
+    if (!loading && urlParams) {
+      makeChosenApisFromParams();
+    }
+  }, [loading, urlParams]);
+
+  const makeChosenApisFromParams = () => {
+    for (const key of urlParams.keys()) {
+      presentedApis.forEach((api: DelegableApi) => {
+        if (api.identifier === key) {
+          dispatch(apiDelegationCheck(api));
+        }
+      });
+    }
+  };
+
+  const addApiToParams = (api: DelegableApi) => {
+    urlParams.append(api.identifier, '');
+    setUrlParams(urlParams);
+  };
+
+  const handleRemove = (api: DelegableApi) => {
+    removeApiFromParams(api);
+    dispatch(filter(filters));
+    dispatch(search(searchString));
+  };
+
+  const removeApiFromParams = (api: DelegableApi) => {
+    urlParams.delete(api.identifier);
+    setUrlParams(urlParams);
+    dispatch(softRemoveApi(api));
```

Comment:
Very small thing but why is softRemoveApi in the removeApi_FromParams_ method?
I would expect it to be in the handleRemove method, after removeApiFromParams has finished.